baseapp-profiles/baseapp_profiles/models.py

```diff
@@ -145,7 +145,7 @@ class ProfilableModel(models.Model):
     profile = models.OneToOneField(
         swapper.get_model_name("baseapp_profiles", "Profile"),
         related_name="%(class)s",
-        on_delete=models.CASCADE,
+        on_delete=models.PROTECT,
         verbose_name=_("profile"),
         null=True,
         blank=True,
```

Comment:
If we're supporting a `ProfilableModel` to _not_ have a profile, why are we not letting them delete said profile once it's been created?  I'm usually suspicious of protect delete behavior on a nullable foreign key, would `SET_NULL` not be more appropriate?

Reply:
Ooooh I thought we wanted all users to have profile. If not, I agree with you.

Up to you @nossila @nick-prat @vitorguima decide what is better for us.

Reply:
@nick-prat its nullable because we were working on projects that already had Users, we could remove the null now if desired.

@rafaelhdr are you looking into having users with profile and without profile mixed up in the same project? I also think that `SET_NULL` might be more appropriate in that case. But the initial idea was to really have a profile for every user, in case that `CASCADE` is fine and easier to handle deletes, but I also like the idea of `PROTECT` that would make you always delete the users and not the profile. We can even remove DELETE option on django admin for Profiles. What you think?

Reply:
@nick-prat @nossila since we need a definition for this let's go with the `SET_NULL` then? So we don't touch the nullable logic and we can keep the package more flexible. I also don't see any big counter points for not using `SET_NULL`.

Reply:
> are you looking into having users with profile and without profile mixed up in the same project?
Nope. But I think you know the roadmap better than me

> we could remove the null now if desired.
@nossila can we remove the nullable?
I think we would have the egg and chicken issue here, no?
We can't create the `User`, because we don't have the `Profile` yet. And we can't create the `Profile` because we don't have the `User` yet.

So, for now, everyone agrees with `SET_NULL`? And then later we discuss better what would be better?

Reply:
> We can even remove DELETE option on django admin for Profiles.
I think regardless of the decision, this seems a good idea. I believe someone was doing something wrong (to delete an user profile) and this Pull Request just solves the error 500. If we don't want people deleting the Profile, it seems a good idea for me.